app/models/vote.rb

```diff
@@ -0,0 +1,25 @@
+class Vote < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :review
+
+  validates :user, presence: true
+  validates :review, presence: true
+  validates :user_id, uniqueness: {
+    scope: :review_id, message: "User can only vote once per review."
+  }
+
+  def self.user_already_voted(user, review)
+    Vote.where(user_id: user.id, review_id: review)
+  end
```

Comment:
I think the name of this method is a bit deceiving when compared to what it's returning. Based on the name, I'd expect the method to return a boolean representing whether or not the "user already voted". It looks like most places you're calling this method, you're calling `.empty?` on it right away, so you could get rid of needing to do that by having this method return a boolean.